### ISSUE TITLE
GObject.Property and setter decorator mistakenly issues reportRedeclaration

### ISSUE THREAD

**Issue by ltsdw** [microsoft/pyright#8941]
**Describe the bug**
While using the `property` and `setter` decorator works normally, using `GObject.Property` decorator and `setter` decorator, pyright issues a reportRedeclaration error message.

**Code or Screenshots**

```python
from gi import require_version
require_version("GObject", "2.0")
from gi.repository.GObject import Object, Property

class Foo(Object):
    def __init__(
        self,
    ) -> None:
        super().__init__()

        self._index: str = "1"


    @Property(type=str, default="")
    def index(self) -> str:
        return self._index


    @index.setter
    def index(self, value: str) -> None:
        self._index = value
```

Third party packages installed:
```
gobject 0.1.0
PyGObject 3.48.2
PyGObject-stubs 2.11.0
```

**VS Code extension or command-line**
Running as a neovim plugin (but also tested using pyright extension on vscode as I reported here https://github.com/fannheyward/coc-pyright/issues/1107 same happens there):

```
[coc-nvim] coc-pyright 1.1.377 with Pyright 1.1.378
```

**Comment by erictraut** [comment]
This behavior is expected, so I don't consider it a bug.

Type checkers like pyright and mypy contain significant special casing to support the stdlib `property`. Such special casing isn't feasible for third-party libraries that attempt to work similarly to `property`. One of the areas that's special-cased is the suppression of a redeclaration error. 

If you want to use this pattern with static type checking, here are a few options:
1. Manually suppress the redeclaration error using `# type: ignore` or `# pyright: ignore`.
2. Disable the `reportRedeclaration` rule for your project.
3. Avoid the redeclaration by using different names for the two methods:

```python
    @Property(type=str, default="")
    def index_(self) -> str:
        return self._index

    @index_.setter
    def index(self, value: str) -> None:
        self._index = value
```
